### PR TITLE
Is condtional style active

### PR DESCRIPTION
**Pull request by Collie-IT**
This is:

```
- [ ] a bugfix
- [x] a new feature

```

Checklist:

- [x] Changes are covered by unit tests
- [x] Code style is respected
- [x] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [x] CHANGELOG.md contains a short summary of the change
- [x] Documentation is updated as necessary -not needed

### Why this change is needed?
PDF (tcpdf) printing does not respect condtional formatting  #1375.  It is necessary to have a function to check if is there an active conditional formatting and  then to set the cell style appropriatly.

### Limitations
All CELLIS condtion type with there operater types/functions are covered. The text operators (STARTSWITH, ENDSWITH, CONTAINS AND NOT CONTAINS) aren't covered because they are currently not implemented in Worksheet.php writeCFRule.

### CODE REVIEW

**Comment by Collie-IT** [comment]
Dear @PowerKiKi, I added your feedback

> $multuseCol and $multuseRow are not explained, and their name are not self-describing

- I changed the names

>     don't use binary file for test, and instead create spreadsheet in memory from scratch

- I added a new test file 
- I added a in memory method because of that excel can do some strange things in the conditional formatting I don't delete the excel test case with a real excel.

Excel manages to have conditions that looks like for e.g. =$E$5:$E$14 NOT Between -$D$5-0.4 And $D$5+0.4 That means the condition will return -$D$5-0.4 and $D$5+0.4 everytime, but this must be adjusted in case of cell $E$6 to -$D$6-0.4 And $D$6+0.4  

> the new method should be useful somewhere in PhpSpreadsheet, otherwise there is no value to add it. The only place I can think of would be in Writer/Html, so Html and PDFs would support conditional styles.

- Indeed there should be an integration to the writer. I would prefer the html writter, too.  Can you suggest a good place? 
- For Tcpdf I have already implemented the code

> to be really useful, we should also have a new Worksheet::getStyle

- Can you tell me how this should look like? I don't understand what I should do in this case.

>  why did you choose bccomp, instead of standard operators ?

- bccomp is used for floats comparison with a precision in number of decimalplaces. (Excel rounds to displayed precision before comparision)

> Do not disable the cache by default, it would be a performance hit. Instead do nothing at all, and let end-users do that manually if they really requires it

- I have excluded the disable cache by default
- But there is an issue at the second condtion. Every time the value of the first condition will be given back from the cache, e.g. condtion1 = 17.5 and condtion2 = -17.5 the return is everytime 17.5. 

I strongly suggest to use   $calcer->disableCalculationCache(); or find the reason for wrong cache return.

File: src/PhpSpreadsheet/Style/Conditional.php [hunk 1]
```diff
@@ -271,4 +274,148 @@ public function __clone()
...
+                    $return = (!(bccomp($calcVal, $coords0Val, $precision) <= 0 || bccomp($calcVal, $coords1Val, $precision) <= 0));
+                } else {
+                    $return = (!(bccomp($calcVal, $coords1Val, $precision) <= 0 || bccomp($calcVal, $coords0Val, $precision) <= 0));
+                }
+
+                break;
+            case '':
+                break;
+            default:
+                throw new \Exception('Unknown ' . $this->getOperatorType() . ' during active check of conditional formatting.');
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Style/Conditional.php:355]:
```suggestion
                throw new \Exception('Unknown ' . $this->getOperatorType() . ' during check of active conditional formatting.');
```

File: src/PhpSpreadsheet/Style/Conditional.php [hunk 2]
```diff
@@ -271,4 +274,148 @@ public function __clone()
...
+     * @param string $cellName
+     * @param int $precision
+     * @param int $updateColumnBy
+     * @param int $updateRowBy
+     *
+     * @throws Exception
+     *
+     * @return bool
+     */
+    public function isActive(Calculation $calcer, Cell $cell, int $precision = 8, int $updateColumnBy = 0, int $updateRowBy = 0)
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Style/Conditional.php:299]:
```suggestion
    public function isActive(Calculation $calcer, Cell $cell, int $precision = 8, int $updateColumnBy = 0, int $updateRowBy = 0): bool
```

**Comment by oleibman** [comment]
Conditional formatting for html (and therefore pdf) was added by PR #4412. Removing stale label, leaving issue closed.

**Pull request closed without merging**